faucet/valve_util.py

```diff
@@ -91,6 +91,16 @@ def get_setting(name):
     """Returns value of specified configuration setting."""
     default_value = DEFAULTS[name]
     result = os.getenv(name, default_value)
+    #split on ':' and find the first suitable path
+    if isinstance(result, str) and isinstance(default_value, str) and not isinstance(default_value, bool):
```

Comment:
There are a few logic flaws in this function (for example it will split an ipv6 address into multiple "paths"). We propose a rework for this function:

1) Let's add an optional argument to get_setting() which lets us ask to "resolve a search path" and then we can modify in valve_ryuapp.py and dp.py when we request a CONFIG or FAUCET_PIPELINE_DIR that we use the "resolve a search path" option.